ports/nrf/mpconfigport.h

```diff
@@ -42,26 +47,108 @@
 #define MICROPY_PY_UBINASCII                     (1)
 #define MICROPY_PY_UJSON                         (1)
 
-// TODO this is old BLE stuff
-#if BLUETOOTH_SD
-    #define MICROPY_PY_BLEIO                     (1)
-    #define MICROPY_PY_BLE_NUS                   (0)
-#else
-    #ifndef MICROPY_PY_BLEIO
-        #define MICROPY_PY_BLEIO                 (0)
-    #endif
-#endif
-
 // 24kiB stack
 #define CIRCUITPY_DEFAULT_STACK_SIZE            0x6000
 
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// This also includes mpconfigboard.h.
 #include "py/circuitpy_mpconfig.h"
 
+// Definitions that might be overriden by mpconfigboard.h
+
+#ifndef CIRCUITPY_INTERNAL_NVM_SIZE
+#define CIRCUITPY_INTERNAL_NVM_SIZE (8192)
+#endif
+
 #ifndef BOARD_HAS_32KHZ_XTAL
 // Assume crystal is present, which is the most common case.
 #define BOARD_HAS_32KHZ_XTAL (1)
 #endif
 
+#if INTERNAL_FLASH_FILESYSTEM
+  #ifndef CIRCUITPY_INTERNAL_FLASH_FILESYSTEM_SIZE
+    #define CIRCUITPY_INTERNAL_FLASH_FILESYSTEM_SIZE (256*1024)
+  #endif
+#else
+  #define CIRCUITPY_INTERNAL_FLASH_FILESYSTEM_SIZE (0)
+#endif
+
+// Flash layout, starting at 0x00000000
+//
+// SoftDevice
+// ISR
+// firmware
+// internal CIRCUITPY flash filesystem (optional)
```

Comment:
Originally I had it just below the bootloader, but I realized that would break any existing on-chip filesystems. We don't have such boards, but there are third-party boards that use on-chip filesystems. The same is true of the atmel-samd layout: I didn't want to destroy all the on-chip filesystems with this upgrade.

We could still decide to do this, but we need **BIG WARNINGS**.

Reply:
As discussed on Discord. Please add a hard coded check to make sure the location of the filesystem is preserved.

Reply:
In atmel-samd, the filesystem is currently below nvm and the USB clock calibration region. So the location of the filesystem depends on the board's choice for those region sizes, which can vary. So there is no absolute location to check against, unless I put that value in every mpconfigboard file. 

I can make nrf be more consistent (just below the bootloader), though it still depends on the filesystem size.

Reply:
That's fine. Let me know when I should take another look at this.